From 35423f29e270bc63b788e6b70c736fbd1467dad1 Mon Sep 17 00:00:00 2001
From: Mike Sul <mike.sul@foundries.io>
Date: Thu, 15 Jun 2023 13:43:51 +0200
Subject: [PATCH] Don't store signatures if none is present

Currently, the copy command prints the message 'Storing signatures' and
calls the signature storing function, even if there are no signatures
present. This can mislead users and make them believe that there are
image signatures.

The proposed change modifies the copy function to print the message and
invoke the image storing function only if there is at least one signature.

Upstream-Status: Pending

Signed-off-by: Mike <mike.sul@foundries.io>
---
 copy/copy.go | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/vendor/github.com/containers/image/v5/copy/copy.go b/vendor/github.com/containers/image/v5/copy/copy.go
index 26521fe0..560b6c1a 100644
--- a/vendor/github.com/containers/image/v5/copy/copy.go
+++ b/vendor/github.com/containers/image/v5/copy/copy.go
@@ -810,9 +810,11 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 	}
 	sigs = append(sigs, newSigs...)
 
-	c.Printf("Storing signatures\n")
-	if err := c.dest.PutSignaturesWithFormat(ctx, sigs, targetInstance); err != nil {
-		return nil, "", "", fmt.Errorf("writing signatures: %w", err)
+	if len(sigs) > 0 {
+		c.Printf("Storing signatures\n")
+		if err := c.dest.PutSignaturesWithFormat(ctx, sigs, targetInstance); err != nil {
+			return nil, "", "", fmt.Errorf("writing signatures: %w", err)
+		}
 	}
 
 	return manifestBytes, retManifestType, retManifestDigest, nil
-- 
2.40.1

